rules/aip0123/aip0123.go

```diff
@@ -76,6 +78,91 @@ func getVariables(pattern string) []string {
 	return answer
 }
 
+// isTopLevelResourcePattern determines if the given pattern is that of a
```

Comment:
While this is syntactically correct, I think we often treat things like "project/{project}/locations/{location}" as *effectively* top-level. I don't know whether that's going to be relevant here, but I thought I'd mention it.

Reply:
I see what you are saying. Let me see if I can come up with a better name...just want to see if the pattern indicates if it has a parent or not by detecting if there are more than two segments (where odd # of segments is a singleton, and event # >= 4 is _technically_ a nested collection)

Reply:
Yup, that feels clearer to me.